### PR TITLE
Lens model cosmology refactoring

### PR DESCRIPTION
**Pull request by sibirrer**
This PR adds a definition in LensModel to update the cosmology instance. The idea is that the API should support all downstream cosmology definitions and serves as a single point to update the cosmology of the lens model configuration.

### CODE REVIEW

File: test/test_LensModel/test_lens_model.py [hunk 1]
```diff
@@ -354,6 +354,62 @@ def test_change_source_redshift(self):
...
+        from astropy.cosmology import FlatwCDM
+
+        cosmo = FlatwCDM(H0=67, Om0=0.3, w0=-0.8)
+        cosmo_new = FlatwCDM(H0=73, Om0=0.3, w0=-1)
+
+        z_lens = 0.5
+        z_source_convention = 2
+        z_source_new = 1
+        kwargs_lens = [{"theta_E": 1, "center_x": 0, "center_y": 0}]
+        # multi-plane lens model
```
Comment (martin-millon) [inline test/test_LensModel/test_lens_model.py:367]: This seems to have only one plane. Can you add a test with two lens planes ?

**Review by martin-millon — APPROVED** [review]
This looks good, thanks @sibirrer ! I'll try to run my tests with this changes,

**Comment by martin-millon** [comment]
are you planning to have the modifications of the Param class to allow the sampling of the cosmological parameter in this PR as well, or should we already merge this?

**Comment by sibirrer** [comment]
I think we can merge this one @martin-millon. I will open a new PR for the parameter class

**Pull request merged**